### PR TITLE
feat: cli: add component installation and update

### PR DESCRIPTION
**Pull request by lizzy-0323**
## Summary

refer to #131 

## Solution Description
Add cli installation and update, currently support ob-operator, ob-dashboard, cert-manager, local-path-provisioner

```bash
make cli-build
```

### CODE REVIEW

**Review by powerfooI — COMMENTED** [review]
Is this PR WIP? If so, you'd better mark it as a draft. :)

**Comment by powerfooI** [comment]
It seems that this PR has a lot of commits that belong to previous PR #496. You could rebase this branch onto master branch to simplify the commit history.

File: internal/assets/cli-templates/component_config.yaml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # LATEST_VERSION
 components:
   cert-manager: "2.2.2_release"
-  # local-path-provisioner: "v0.0.28"
-  # oceanbase-dashboard: "0.2.1"
+  local-path-provisioner: "v0.0.28"
+  local-path-provisioner-dev: "master"
+  ob-dashboard: "0.3.0"
   ob-operator: "2.2.2_release"
+  ob-operator-dev: "master"
```
Comment (powerfooI) [inline internal/assets/cli-templates/component_config.yaml:8]: `ob-operator-dev` and `local-path-provisioner-dev` should be removed.

Comment (lizzy-0323) [reply internal/assets/cli-templates/component_config.yaml:8]: `dev` means development  version of components, maybe they should be reserved to provide more choice.

**Comment by lizzy-0323** [comment]
> Is this PR WIP? If so, you'd better mark it as a draft. :)

install and update components has been implement.

File: internal/cli/cmd/install/install.go [hunk 2]
```diff
@@ -25,15 +25,32 @@ func NewCmd() *cobra.Command {
...
+- cert-manager: Creates TLS certificates for workloads in Kubernetes and renews the certificates before they expire, ob-operator relies on it for certificate management, which should be installed beforehand.
+
+if not specified, install all the components`,
 		PreRunE: o.Parse,
 		Args:    cobra.MaximumNArgs(1),
 		Run: func(cmd *cobra.Command, args []string) {
-			if err := o.Install(); err != nil {
-				logger.Fatalln(err)
+			if len(args) == 0 {
+				logger.Println("Install all the components")
```
Comment (chris-sun-star) [inline internal/cli/cmd/install/install.go:43]: I think it's better to just install the ob-operator and ob-dashboard by default, local-path-provisioner and cert-manager are both 3rd party dependencies, these may already exist on the user's environment.

File: internal/cli/install/install.go [hunk 3]
```diff
@@ -56,30 +55,59 @@ func (o *InstallOptions) Parse(_ *cobra.Command, args []string) error {
...
 			return err
 		}
+		versionFlag := fmt.Sprintf("--version=%s", version)
+		cmd = exec.Command("helm", "install", "oceanbase-dashboard", "ob-operator/oceanbase-dashboard", versionFlag)
+	}
+	return runCmd(cmd)
+}
+
+// preRunCmd preRun two commands for ob-dashboard installation
+func preRunCmd() error {
```
Comment (chris-sun-star) [inline internal/cli/install/install.go:87]: Maybe this function can have a better name, this function is just called when installing dashboard, maybe 'updateHelmRepo' is a better choice

File: internal/cli/cmd/install/install.go (hunk 2 above)
Comment (powerfooI) [reply internal/cli/cmd/install/install.go:43]: That's true. `local-path-provisioner` is a weak dependency, and you could check the installation of `cert-manager` by checking existence of resources in `cert-manager.io/v1` like following,

image

Comment (lizzy-0323) [reply internal/cli/cmd/install/install.go:43]: Ok,  Next, I will only provide installation of ob related components by default.

File: internal/cli/install/install.go (hunk 3 above)
Comment (lizzy-0323) [reply internal/cli/install/install.go:87]: Ok, it has been changed.

File: internal/cli/install/install.go [hunk 4]
```diff
@@ -39,75 +44,136 @@ func (o *InstallOptions) AddFlags(cmd *cobra.Command) {
...
+	cmd := exec.Command("kubectl", "get", "crds", "-o", "name", "|", "grep", "cert-manager")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return false
+	}
+
+	// Check if the output contains cert-manager resources
+	expectedResources := []string{
```
Comment (powerfooI) [inline internal/cli/install/install.go:118]: What if a different version of cert-manager has different CRDs?

File: internal/cli/update/update.go [hunk 5]
```diff
@@ -18,16 +18,16 @@ import (
 
 	"github.com/spf13/cobra"
 
-	"github.com/oceanbase/ob-operator/internal/cli/utils"
+	"github.com/oceanbase/ob-operator/internal/cli/install"
 )
 
 type UpdateOptions struct {
```
Comment (powerfooI) [inline internal/cli/update/update.go:24]: How about `type UpdateOptions install.InstallOptions`

File: internal/cli/install/install.go (hunk 4 above)
Comment (lizzy-0323) [reply internal/cli/install/install.go:118]: Fine，maybe I still need  to  run `kubectl api-resources | grep cert-manager.io/v1` to check the environment?

Comment (powerfooI) [reply internal/cli/install/install.go:118]: Actually what I mean is that you require all CRDs in the list appear in the output of kubectl command. Would that be too strict?

Comment (lizzy-0323) [reply internal/cli/install/install.go:118]: In their official documentation:
`kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.15.3/cert-manager.yaml
`should be run to install all the CRDs.
I think if the user installs this way, maybe all CRDs should be found in the environment?

Comment (powerfooI) [reply internal/cli/install/install.go:118]: If the version is fixed to v1.15.3, the requirement is proper. How about other version like v1.14, v1.13 or upcoming v1.16? The list of CRDs should be check if they are the same between versions.

Comment (lizzy-0323) [reply internal/cli/install/install.go:118]: I have tried official repo with version of v1.14, v1.13, v1.12 and v1.11, the list of CRDs are the same.

Comment (powerfooI) [reply internal/cli/install/install.go:118]: That's great!

**Review by powerfooI — APPROVED** [review]
LGTM

**Pull request merged**